test/performance/scale_test.go

```diff
@@ -0,0 +1,153 @@
+// +build performance
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// scale_test.go brings up a number of services tracking the time to various waypoints.
+
+package performance
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/test-infra/shared/testgrid"
+
+	"github.com/knative/serving/test/e2e"
+)
+
+type metrics struct {
+	min           time.Duration
+	max           time.Duration
+	totalDuration time.Duration
+	num           int64
+}
+
+func (m *metrics) Add(start time.Time) {
+	// Compute the duration since the provided start time.
+	d := time.Since(start)
+
+	if d < m.min || m.min == 0 {
+		m.min = d
+	}
+	if d > m.max {
+		m.max = d
+	}
+	m.totalDuration += d
+	m.num++
+}
+
+func (m metrics) Min() float32 {
+	return float32(m.min.Seconds())
+}
+
+func (m metrics) Max() float32 {
+	return float32(m.max.Seconds())
+}
+
+func (m metrics) Avg() float32 {
+	return float32(m.totalDuration.Seconds()) / float32(m.num)
```

Comment:
check for num>0?

Reply:
the only way metrics are created is by the defaulting in `Add()` below, which immediately calls `Add()` above, which always makes `m.num > 0`.

I'd rather avoid the `error` return value (and assorted error checking)

Reply:
you can just return 0 :)

Reply:
I went with `-1.0`, which should be impossible.